src/app/App.jsx

```diff
@@ -0,0 +1,32 @@
+import { Suspense, lazy } from 'react';
+import { Provider } from 'react-redux';
+import { BrowserRouter } from 'react-router-dom';
+import { AppProvider, ThemeProvider } from './providers';
+import Layout from './layout';
+
+const PublicRoutes = lazy(() => import('pages/public'));
+const PrivateRoutes = lazy(() => import('pages/private'));
+
+const App = ({ store }) => (
+  <Provider store={store}>
+    <BrowserRouter>
+      <ThemeProvider>
+        <AppProvider>
+          {(isLoggedIn) =>
+            isLoggedIn ? (
+              <Layout>
+                <PrivateRoutes />
+              </Layout>
+            ) : (
+              <Suspense fallback={null}>
+                <PublicRoutes />
```

Comment:
Do public routes require any sort of Layout too ?

Reply:
Agree. Fixed